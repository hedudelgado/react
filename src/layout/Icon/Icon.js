import React from 'react';
import { string, oneOf } from 'prop-types';
import classNames from 'classnames';

const alignments = {
  left: 'pl1',
  right: 'pr1',
};

const Icon = ({ icon, push, className }) => {
  const cssClasses = classNames(
    `wb-icon_${icon}`,
    alignments[push],
    className,
  );

  return (
    <i className={cssClasses} />
  );
};

Icon.propTypes = {
  icon: string.isRequired,
  push: oneOf(Object.keys(alignments)),
  className: string,
};

export default Icon;
