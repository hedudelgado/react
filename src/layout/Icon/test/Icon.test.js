import React from 'react';
import { shallow } from 'enzyme';
import Icon from '..';

describe('Icon component', () => {
  it('should render as expected', () => {
    expect(shallow(<Icon icon="foo" push="right" />)).toMatchSnapshot();
  });

  it('should not have padding if push prop is undefined', () => {
    const wrapper = shallow(<Icon icon="foo" />);

    expect(wrapper).not.toHaveProp('push');
    expect(wrapper).not.toHaveClassName('pl1');
    expect(wrapper).not.toHaveClassName('pr1');
  });

  it('should have padding if push prop is set', () => {
    expect(shallow(<Icon icon="foo" push="left" />)).toHaveClassName('pl1');
    expect(shallow(<Icon icon="foo" push="right" />)).toHaveClassName('pr1');
  });
});
