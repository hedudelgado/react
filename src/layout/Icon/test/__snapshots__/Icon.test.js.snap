// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should render as expected 1`] = `
<i
  className="wb-icon_foo pr1"
/>
`;
